### PR TITLE
[Security][Automations] Remove "x-ms-discriminator-value" from Inherited Models

### PR DESCRIPTION
**Pull request by mikreite**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by mikreite** [comment]
removing for failing the GO SDK

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json [hunk 1]
```diff
@@ -497,7 +497,6 @@
     "AutomationActionLogicAppInput": {
       "type": "object",
       "description": "The logic app action that should be triggered.",
-      "x-ms-discriminator-value": "LogicApp",
```
Comment (jhendrixMSFT) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: Is `AutomationActionLogicAppInput` never sent across the wire?

Comment (jhendrixMSFT) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: Same question applies to the event hub types.

Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: AutomationActionLogicApp is kind of an "abstract" model which is derived to two models: AutomationActionLogicAppInput - which is being **sent** for PUT requests, and AutomationActionLogicAppOutput - which is the one that is **returned** for GET and PUT. Actually, there is the "uri" field which is not allowed to be returned for security reasons.
Same approach is for EventHub.

I had tagged the 3 models (abstract + input + output) with the x-ms-discriminator-value property, for the same value ("LogicApp"). Apparently this caused the GO SDK to fail.
I am wondering how the following schema should be presented in swagger though:

(copied from examples)
**PUT Request**:
"actions": [
          {
            "logicAppResourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
            "actionType": "LogicApp",
            "uri": "https://exampleTriggerUri1.com"
          }
        ]
**Response**:
"actions": [
            {
              "logicAppResourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
              "actionType": "LogicApp"
            }
          ]

Comment (jhendrixMSFT) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: If `AutomationActionLogicApp` is a "base class" that is never sent/received then I believe that it should omit the `x-ms-discriminator-value` and that each derived type should define its own `x-ms-discriminator-value` value.  I am by no means an expert in this area though, @fearthecowboy can you please advise?

**Comment by mikreite** [comment]
@jhendrixMSFT and @fearthecowboy hi, I tried to use the "writeOnly" swagger property to avoid the need to use 2 different models (input and output), but apparently this is not supported in Swagger 2.0 (only 3.0).
As a mitigation - I has unified the input+output models (as actually they share all properties - the secret), while stating in the description that this is not being return in any response.
WDYT?
Please remember that no docs have been published yet for this resource.

**Comment by mikreite** [comment]
adding @dafishma

**Comment by mikreite** [comment]
@KrisBash, @njuCZ  ? appreciate your review and approval the soonest, thanks

**Review by pilor — APPROVED** [review]

**Comment by mikreite** [comment]
@njuCZ can u please merge the PR? thanks

**Comment by mikreite** [comment]
@jhendrixMSFT, there is no response from @njuCZ for quite a while. Could you please review and merge this PR?

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**